### DISCUSSION TITLE
Added 'Last Transfer' date in torrents details: list, modal, UI column settings

### DISCUSSION THREAD

**Discussion by gboudreau** [jesec/flood#9] [Ideas]
Type: Discussion

## Discussion

I couldn't send you a PR, because your repo is disconnected from the original one, but maybe you're interested in this:
https://github.com/Flood-UI/flood/pull/882

Cheers.

**Comment by jesec** [comment]
It looks good. However, it could be problematic to support a feature that is not present in vanilla rTorrent and requires user to change their rTorrent configurations. Some users might complain that "Last transfer" is not working if they are not aware of the setup procedures.

**Comment by gboudreau** [comment]
True; would be better to make it user-selectable in the settings, and add a link to the documentation next to it.
If other users are interested, at least now that can see it could work, and might upvote it, which could give us an idea on if it's worth the effort to continue working in that direction.

**Comment by jesec** [comment]
I think you can probably try to send necessary commands to rTorrent from Flood. Another option is to incorporate the feature into rTorrent.

**Comment by jesec** [comment]
qBittorrent (via `last_activity`) and Transmission (via `activityDate`) have the feature.

It is relatively trivial to implement this feature in my fork of rTorrent. 

I will revisit this PR after that.

**Reply by jesec** [reply]
I have implemented the last activity in rTorrent with https://github.com/jesec/libtorrent/commit/5d017ce1934815c05bfab3be688e906b1a607b76 and https://github.com/jesec/rtorrent/commit/c5fdedd54a87b72b13b15fb56443c33b104d6481. The function is fully event-based, and should have a well-defined and stable behavior.

"Activity" here means "last time any peer was connected", consistent with @pyroscope's definition of `d.timestamp.last_active` in https://rtorrent-docs.readthedocs.io/en/latest/cmd-ref.html .

Some caveats: the date would only be recorded when the last peer disconnects naturally. So if the user manually stopped the torrent, or stop the rTorrent when the torrent is still active, the timestamp would not be recorded. This is because when the torrent closes (or download object destructs in C++), pending events are destroyed as well. Additionally, users have to check if the torrent is currently active by themselves, so rTorrent does not have to frequently write current time to the session.

I plan to try to address these caveats in the next revision, but I think the feature is good enough for majority of use cases. I will soon add a `Last Activity` feature to Flood.
